File: c/gemini/p121.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    // Allocate two chunks of size 10 (integers)
    int *chunk1 = (int *)malloc(10 * sizeof(int));
    int *chunk2 = (int *)malloc(10 * sizeof(int));

    // Initialize first chunk with some data
    for(int i = 0; i < 10; i++) chunk1[i] = i + 1;

    // Copy contents of chunk1 into chunk2
    memcpy(chunk2, chunk1, 10 * sizeof(int));

    // Print contents of chunk2
    printf("Chunk 2: ");
    for(int i = 0; i < 10; i++) printf("%d ", chunk2[i]);

    // Free the first chunk
    free(chunk1);
    printf("\nChunk 1 freed. Chunk 2 remains accessible.\n");

    // Print contents of chunk2 again
    printf("Chunk 2: ");
    for(int i = 0; i < 10; i++) printf("%d ", chunk2[i]);

    // Free the second chunk
    free(chunk2);

    return 0;
}

File: javascript/gemini/p121.js
// Allocate two chunks
let chunk1 = Array.from({length: 10}, (_, i) => i + 1);
let chunk2 = new Array(10);

// Copy contents
chunk2 = [...chunk1];

// Print contents
console.log("Chunk 2:", chunk2);

// "Free" the first chunk
chunk1 = null;
console.log("Chunk 1 set to null.");

// Print contents of the second chunk
console.log("Chunk 2:", chunk2);

// "Free" the second chunk
chunk2 = null;